polkadot/node/subsystem-util/src/inclusion_emulator/mod.rs

```diff
@@ -600,6 +600,18 @@ impl Fragment {
 		validation_code_hash: &ValidationCodeHash,
 		persisted_validation_data: &PersistedValidationData,
 	) -> Result<ConstraintModifications, FragmentValidityError> {
+		// Filter UMP signals and the separator.
+		let upward_messages = if let Some(separator_index) =
```

Comment:
would be good to deduplicate this filtering in a function on `CandidateCommitments`. We're doing this in multiple places. we should also use the `UMP_SEPARATOR`

Reply:
Yeah, can you add such a function  to your PR: https://github.com/paritytech/polkadot-sdk/pull/6217  ? You also to do this and it's easier to propagate to this PR since your has master as base.

```
		let upward_messages = upward_messages
			.iter()
			.take_while(|message| message != &&UMP_SEPARATOR)
			.collect::<Vec<_>>();
```